tensorflow_datasets/core/lazy_imports_lib.py

```diff
@@ -21,23 +21,59 @@
 from __future__ import print_function
 
 import importlib
+import sys
+import types
+import imp
 
 from tensorflow_datasets.core.utils import py_utils as utils
 
+_ERR_MSG = ("Failed importing {name}. This likely means that the dataset "
+            "requires additional dependencies that have to be "
+            "manually installed (usually with `pip install {name}`). See "
+            "setup.py extras_require.")
+
+# `lazy_imports` currently only allow the following external dependencies to
+# be used during dataset generation. 
+_ALLOWED_LAZY_DEPS = [
+    "apache_beam",
+    "crepe",
+    "cv2",
+    "h5py",
+    "langdetect",
+    "librosa",
+    "matplotlib",
+    "mwparserfromhell",
+    "nltk",
+    "pandas",
+    "PIL_Image",
+    "pretty_midi",
+    "pydub",
+    "scipy",
+    "skimage",
+    "tensorflow_io",
+    "tldextract",
+    "os",
+    # "test_foo",
+]
+
 
 def _try_import(module_name):
   """Try importing a module, with an informative error message on failure."""
   try:
     mod = importlib.import_module(module_name)
     return mod
   except ImportError:
-    err_msg = ("Failed importing {name}. This likely means that the dataset "
-               "requires additional dependencies that have to be "
-               "manually installed (usually with `pip install {name}`). See "
-               "setup.py extras_require.").format(name=module_name)
+    err_msg = _ERR_MSG.format(name=module_name)
     utils.reraise(suffix=err_msg)
 
 
+class FakeModule(types.ModuleType):
+  """Create a fake module and raise ImportError whenever a attribute is accessed""" 
+  def __getattribute__(self, _):
+    err_msg = _ERR_MSG.format(name=self.module_name)
+    raise ImportError(err_msg)
```

Comment:
I think `return super(FakeModule, self).__getattribute__(_)` can avoid infinite recursion error

Reply:
That won't be required as I am already raising an `ImportError`.
Program execution stops as soon as an exception is encountered.

Reply:
Okay Try this  replace `self.module_name` by `_name ` where `_name=` object.__getattribute__(self, 'module_name') `

Reply:
Thanks for the help.
I used `_getattribute__` function instead of `_getattr__`.
Issue resolved. :smile: